lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -136,6 +139,405 @@ def test_alreadyrunningerror_ajax(self):
+            self.assertEqual(student['email'], user.email)
+        success = json.loads(self._delete_temp_query(query_id).content)['success']
+        self.assertEquals(success, True)
+        temp = self._get_temp_query_ids()
+        self.assertEquals(len(temp), 0)
+        self.assertEquals(len(StudentsForQuery.objects.all()), 0)
+        return query_id
+
+    def _save_query(self, existing):
+        """
+        Save a group query
+        """
+
+        save_url = reverse('save_query', kwargs={'course_id': self.course_key})
+        save_args = {
+            'existing': ','.join(existing),
+        }
+        response = self.client.get(save_url, save_args)
+        return response
+
+    def _delete_saved_query(self, query_id):
+        """
+        Deletes a group query
+        """
+
+        delete_url = reverse('delete_saved_query', kwargs={'course_id': self.course_key})
+        use_delete_url = '/'.join([delete_url, str(query_id)])
+        response = self.client.get(use_delete_url)
+        return response
+
+    def _get_saved_queries(self):
+        """
+        Retrieve saved group queries
+        """
+
+        get_saved_url = reverse('get_saved_queries', kwargs={'course_id': self.course_key})
+        temp_response = self.client.get(get_saved_url)
+        queries = json.loads(temp_response.content)['queries']
+        return queries
+
+    def test_make_open_query(self):
+        """
+        Ensures the logic is sound for "opened" students
+        """
+
+        self._get_query_results(self.OPEN)
+
+    def test_make_closed_query(self):
+        """
+        Ensures the logic is sound for "completed" students
+        """
+
+        self._get_query_results(self.COMPLETED)
+
+    def test_make_not_opened_query(self):
+        """
+        Ensures the logic is sound for "not opened" students
+        """
+
+        self._get_query_results(self.NOT_OPEN)
+
+    def test_make_not_completed_query(self):
+        """
+        Ensures the logic is sound for "not completed" students
+        """
+
+        self._get_query_results(self.NOT_COMPLETED)
+
+    def test_make_all_query(self):
+        """
+        Ensures the logic is sound for aggregating students
+        """
+
+        self.assertEquals(len(self._get_temp_query_ids()), 0)
+        self._make_query(self.OPEN)
+        self._make_query(self.NOT_COMPLETED)
+        queries, students = self._get_query_students()
+        self._check_against_students(students, [self.open_user, self.completed_user, self.not_opened_user])
+        self._delete_bulk_temp_query(queries)
+        self.assertEquals(len(self._get_temp_query_ids()), 0)
+        self._make_query(self.OPEN, joining="AND")
+        self._make_query(self.NOT_COMPLETED, joining="AND")
+        queries2, students2 = self._get_query_students()
+        csv_string = self._get_query_students(csv=True)
+        to_match = '"email, name"\r\n"%s","%s"\r\n' % (self.open_user.email, self.open_user.profile.name)
```

Comment:
Throughout the code base, we use the `str`'s `format` method, instead of `%`-based interpolation.

```python
'"email, name"\r\n"{email}","{name}"\r\n'.format(
    email='test@example.com',
    name='test',
)